src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -86,19 +86,36 @@ const Map<wifi_iface_state, WiFiManager::StationStatus, 10>
                               { WIFI_STATE_COMPLETED, WiFiManager::StationStatus::FULLY_PROVISIONED } });
 
 const Map<uint32_t, WiFiManager::NetEventHandler, 4>
-    WiFiManager::sEventHandlerMap({ { NET_EVENT_WIFI_SCAN_RESULT, WiFiManager::ScanResultClbk },
-                                    { NET_EVENT_WIFI_SCAN_DONE, WiFiManager::ScanDoneClbk },
-                                    { NET_EVENT_WIFI_CONNECT_RESULT, WiFiManager::ConnectClbk },
-                                    { NET_EVENT_WIFI_DISCONNECT_RESULT, WiFiManager::DisconnectClbk } });
+    WiFiManager::sEventHandlerMap({ { NET_EVENT_WIFI_SCAN_RESULT, WiFiManager::ScanResultHandler },
+                                    { NET_EVENT_WIFI_SCAN_DONE, WiFiManager::ScanDoneHandler },
+                                    { NET_EVENT_WIFI_CONNECT_RESULT, WiFiManager::ConnectHandler },
+                                    { NET_EVENT_WIFI_DISCONNECT_RESULT, WiFiManager::DisconnectHandler } });
 
-void WiFiManager::WifiMgmtEventHandler(NetEventCallback * cb, uint32_t mgmtEvent, struct net_if * iface)
+void WiFiManager::WifiMgmtEventHandler(NetEventCallback * cb, uint32_t mgmtEvent, NetInterfacePtr iface)
 {
     if (0 == strcmp(iface->if_dev->dev->name, "wlan0"))
     {
-        DeviceLayer::SystemLayer().ScheduleLambda([cb, mgmtEvent] { sEventHandlerMap[mgmtEvent](cb); });
+        NetworkEventData * eventData = new NetworkEventData;
```

Comment:
This could be avoided - you could pass contents of this structure to a lambda passed to `ScheduleLambda` and get rid off one `reinterpret_cast` at least. But if you prefer it this way please use `Platform::UniquePtr` not to leak memory on failures.

Reply:
Using `ScheduleLambda` was my first idea, Please see the comment above.

Reply:
We can still remove `NetworkEventData` and pass info as an allocated buffer, right?